api/src/main/java/com/datastrato/gravitino/file/FileCatalog.java

```diff
@@ -0,0 +1,93 @@
+import com.datastrato.gravitino.exceptions.NoSuchSchemaException;
+import java.util.Map;
+
+/**
+ * The FileCatalog interface defines the public API for managing file objects in a schema. If the
+ * catalog implementation supports file objects, it should implement this interface.
+ */
+public interface FileCatalog {
+
+  /**
+   * List the files in a schema namespace from the catalog.
+   *
+   * @param namespace A schema namespace.
+   * @return An array of file identifiers in the namespace.
+   * @throws NoSuchSchemaException If the schema does not exist.
+   */
+  NameIdentifier[] listFiles(Namespace namespace) throws NoSuchSchemaException;
+
+  /**
+   * Load file metadata by {@link NameIdentifier} from the catalog.
+   *
+   * @param ident A file identifier.
+   * @return The file metadata.
+   * @throws NoSuchFileException If the file does not exist.
+   */
+  File loadFile(NameIdentifier ident) throws NoSuchFileException;
+
+  /**
+   * Check if a file exists using an {@link NameIdentifier} from the catalog.
+   *
+   * @param ident A file identifier.
+   * @return true If the file exists, false otherwise.
+   */
+  default boolean fileExists(NameIdentifier ident) {
+    try {
+      loadFile(ident);
+      return true;
+    } catch (NoSuchFileException e) {
+      return false;
+    }
+  }
+
+  /**
+   * Create a file metadata in the catalog.
+   *
+   * @param ident A file identifier.
+   * @param comment The comment of the file.
+   * @param format The format of the underlying filesets that is managed by this file object.
+   * @param properties The properties of the file.
+   * @return The created file metadata
+   * @throws NoSuchSchemaException If the schema does not exist.
+   * @throws FileAlreadyExistsException If the file already exists.
+   */
+  File createFile(
+      NameIdentifier ident, String comment, File.Format format, Map<String, String> properties)
+      throws NoSuchSchemaException, FileAlreadyExistsException;
+
+  /**
+   * Apply the {@link FileChange change} to a file in the catalog.
+   *
+   * <p>Implementation may reject the change. If any change is rejected, no changes should be
+   * applied to the file.
+   *
+   * @param ident A file identifier.
+   * @param changes The changes to apply to the file.
+   * @return The altered file metadata.
+   * @throws NoSuchFileException If the file does not exist.
+   * @throws IllegalArgumentException If the change is rejected by the implementation.
+   */
+  File alterFile(NameIdentifier ident, FileChange... changes)
+      throws NoSuchFileException, IllegalArgumentException;
+
+  /**
+   * Drop a file from the catalog.
+   *
+   * <p>The underlying files will be deleted if this file is managed, otherwise, only the metadata
+   * will be dropped.
+   *
+   * @param ident A file identifier.
+   * @return true If the file is dropped, false the file did not exist.
+   */
+  boolean dropFile(NameIdentifier ident);
```

Comment:
Do we need the api `purgeFile`? It may be a little weird to  use the name `purge`. You should get the point of mine. It means that we can delete the file of external system.

Reply:
No. I don't plan to add purgeFile API, this seems not so necessary for now. We can add it when we really have this requirement.

Reply:
> This API will drop all files underlying the path, right? Do we support the subdirectory? If not, how can we deal with the TTL? for example I want to drop the files in `/metalake/catalog/schema/my_file/date=20201010`

Yes, we can support it using FS API as I mentioned in the design doc.